Eastern-Research-Group/mywaterway
# Feature/handle script tags in route

## Related Issues:
* [https://app.breeze.pm/projects/100762/cards/3316378](https://app.breeze.pm/projects/100762/cards/3316378)

## Main Changes:
* Updated the reach-router code to check for `<script>` tags in the url. If the url contains a `<script>` tag they will be rerouted to the new `invalid-url` error page. 
  * I couldn't find a good way to handle this situation. If I threw an error or set an error state value, the error boundary would be displayed, but the styles got messed up due to the script tag in the url. The only reliable way I found to display an error with the correct styles was to navigate them to a new error page and call `window.location.reload()` to refresh the page. 
* Updated the location search component to check for `<script>` tags and display an error message above the component.
  * I handled this case separately so the user wouldn't have to go back to the homepage and then try another search. Handling the error here allows them to just change the search and try again.
  * I ended up moving the text above the search input inside of this component. I did this so the error message on the community page would be positioned in the same place as the location is not available error message.

## Steps To Test:
1. Navigate to [http://localhost:3000/community/%3Cscript%3Ealert('brad')%3C/script%3E/overview](http://localhost:3000/community/%3Cscript%3Ealert('brad')%3C/script%3E/overview) and make sure the new "Sorry, but the url entered was invalid" error message screen is displayed.
2. Go to the home page and search for `<script>` and verify the `Invalid search. Please try a new search.` error message is displayed.
3. Redo the same search tests on the "Explore Topics" search screens.
4. Run the cypress tests and verify they work.
